### PR TITLE
- Remove 180D EMAs

### PR DESCRIPTION
**Pull request by grandizzy**
- Store lupFactor = curent lup / inflator for borrowers on borrower interactions with the pool
- Update tests

### CODE REVIEW

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/erc20/ERC20Pool.sol [hunk 1]
```diff
@@ -395,13 +401,14 @@ contract ERC20Pool is IERC20Pool, ScaledPool {
...
 
-    function borrowerInfo(address borrower_) external view override returns (uint256, uint256, uint256, uint256) {
+    function borrowerInfo(address borrower_) external view override returns (uint256, uint256, uint256, uint256, uint256) {
         uint256 pendingDebt = Maths.wmul(borrowers[borrower_].debt, Maths.wdiv(_pendingInflator(), inflatorSnapshot));
 
         return (
             borrowers[borrower_].debt,            // accrued debt (WAD)
             pendingDebt,                          // current debt, accrued and pending accrual (WAD)
             borrowers[borrower_].collateral,      // deposited collateral including encumbered (WAD)
+            borrowers[borrower_].lupFactor,       // LUP / inflator (WAD)
```
Comment (ith-harvey) [inline src/erc20/ERC20Pool.sol]: Would rename this comment to -> `// LUP / inflator, used in neutralPrice calc`

File: src/erc20/interfaces/IERC20Pool.sol [hunk 2]
```diff
@@ -155,12 +156,14 @@ interface IERC20Pool is IScaledPool {
      *  @notice Struct holding borrower related info.
      *  @param  debt             Borrower debt, WAD units.
      *  @param  collateral       Collateral deposited by borrower, WAD units.
+     *  @return lupFactor        The LUP / inflator snapshot factor used.
```
Comment (ith-harvey) [inline src/erc20/interfaces/IERC20Pool.sol]: change to comment above

File: src/erc20/interfaces/IERC20Pool.sol [hunk 3]
```diff
@@ -282,6 +285,7 @@ interface IERC20Pool is IScaledPool {
      *  @return debt_             Borrower accrued debt (WAD)
      *  @return pendingDebt_      Borrower current debt, accrued and pending accrual (WAD)
      *  @return collateral_       Deposited collateral including encumbered (WAD)
+     *  @return lupFactor_        LUP / inflator (WAD)
```
Comment (ith-harvey) [inline src/erc20/interfaces/IERC20Pool.sol]: same as above

File: src/erc721/ERC721Pool.sol [hunk 4]
```diff
@@ -358,13 +365,14 @@ contract ERC721Pool is IERC721Pool, ScaledPool {
...
 
-    function borrowerInfo(address borrower_) external view override returns (uint256, uint256, uint256[] memory, uint256) {
+    function borrowerInfo(address borrower_) external view override returns (uint256, uint256, uint256[] memory, uint256, uint256) {
         uint256 pendingDebt = Maths.wmul(borrowers[borrower_].debt, Maths.wdiv(_pendingInflator(), inflatorSnapshot));
 
         return (
             borrowers[borrower_].debt,                         // accrued debt (WAD)
             pendingDebt,                                       // current debt, accrued and pending accrual (WAD)
             borrowers[borrower_].collateralDeposited.values(), // deposited collateral including encumbered (WAD)
+            borrowers[borrower_].lupFactor,                    // LUP / inflator (WAD)
```
Comment (ith-harvey) [inline src/erc721/ERC721Pool.sol]: rename comment everywhere as above

File: src/_test/ERC20Pool/ERC20ScaledPoolInterestRateAndEMAs.t.sol [hunk 5]
```diff
@@ -216,4 +218,135 @@ contract ERC20ScaledInterestRateTest is ERC20HelperContract {
         );
     }
 
+    function testScaledPoolEMAAndTargetUtilizationUpdate() external {
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20ScaledPoolInterestRateAndEMAs.t.sol:221]: Like that we have a test for target utilization, could you also include `assertEq()`EMA checks in this test as well?

**Review by ith-harvey — COMMENTED** [review]
Looks good just a couple cleanups

File: src/erc20/ERC20Pool.sol (hunk 1 above)
Comment (grandizzy) [reply src/erc20/ERC20Pool.sol]: changed in https://github.com/ajna-finance/contracts/pull/263/commits/582a145651c311d687d9668f8890237f8dedc480

File: src/erc20/interfaces/IERC20Pool.sol (hunk 2 above)
Comment (grandizzy) [reply src/erc20/interfaces/IERC20Pool.sol]: changed in https://github.com/ajna-finance/contracts/pull/263/commits/582a145651c311d687d9668f8890237f8dedc480

File: src/erc20/interfaces/IERC20Pool.sol (hunk 3 above)
Comment (grandizzy) [reply src/erc20/interfaces/IERC20Pool.sol]: changed in https://github.com/ajna-finance/contracts/pull/263/commits/582a145651c311d687d9668f8890237f8dedc480

File: src/erc721/ERC721Pool.sol (hunk 4 above)
Comment (grandizzy) [reply src/erc721/ERC721Pool.sol]: changed in https://github.com/ajna-finance/contracts/pull/263/commits/582a145651c311d687d9668f8890237f8dedc480

File: src/_test/ERC20Pool/ERC20ScaledPoolInterestRateAndEMAs.t.sol (hunk 5 above)
Comment (grandizzy) [reply src/_test/ERC20Pool/ERC20ScaledPoolInterestRateAndEMAs.t.sol:221]: added in https://github.com/ajna-finance/contracts/pull/263/commits/582a145651c311d687d9668f8890237f8dedc480

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by EdNoepel — APPROVED** [review]
Looks good.  👍

**Pull request merged**